doc/oidc.md

```diff
@@ -0,0 +1,88 @@
+
+#### xAPI Resources
+
+xAPI resources accept tokens with the following scopes:
+
+| Scope              | Capability                                    |
+| ---                | ---                                           |
+| `all`              | Full read/write access to all xAPI resources. |
+| `all/read`         | Read-only access to all xAPI resources.       |
+| `statements/read`  | Read-only access to xAPI Statements           |
+| `statements/write` | Write-only access to xAPI Statements          |
+
+When SQL LRS accepts xAPI statements via OIDC auth it uses token claims to form the xAPI Authority. See [Authority Configuration](authority.md#oidc-authority) for more information.
+
+#### Admin API Resources
+
+Admin API resources share a single scope `admin` that represents full administrative control over SQL LRS.
+
+Administrative functions like credential provisioning require a local admin account. After decoding the token SQL LRS will ensure that an account exists (or is created) with a `username` matching the token's `sub` claim and an `oidc_issuer` matching the `iss` claim. If the user exists but the issuer does not match the request will fail with a 401 status.
+
+##### Local Admins
+
+By default SQL LRS will disable local administrator account usage and management when `LRSQL_OIDC_ISSUER` is provided. To enable local admin accounts when using OIDC, set the `LRSQL_OIDC_ENABLE_LOCAL_ADMIN` (`lrs.oidcEnableLocalAdmin`) variable to `true`.
+
+### Admin UI Authentication
+
+The LRS Admin UI supports interactive login via an OIDC identity provider. To enable this functionality you must provide the OIDC issuer and audience (not optional in this case) as above and additionally set the `LRSQL_OIDC_CLIENT_ID` (`webserver.oidcClientId`) to the Client ID representing your Admin UI.
+
+When enabled the LRS will send configuration to the Admin UI directing it to offer OIDC login. Click the OIDC login link to be redirected to your identity provider for login. Upon a successful login you will be redirected to the Admin UI.
+
+#### Client Template
+
+The LRS Admin UI uses [oidc-client-js](https://github.com/IdentityModel/oidc-client-js) to manage communication with the OIDC identity provider. For some providers it may be necessary to customize client configuration in which case the `LRSQL_OIDC_CLIENT_TEMPLATE` (`webserver.oidcClientTemplate`) variable can be set. Note that the `redirect_uri` and `post_logout_redirect_uri` will be set by the Admin UI client and should not be provided.
+
+### Identity Providers
+
+OIDC support is currently developed and tested against [Keycloak](https://www.keycloak.org/) but may work with other identity providers that implement the specification.
+
+#### Keycloak Demo
+
+This repository contains a Docker Compose file and configuration for a demo instance of keycloak that you can run locally to try out OIDC support. To run keycloak:
+
+    cd dev-resources/keycloak_demo
+    docker compose up
```

Comment:
You could just put the make target here since it does the same thing.

And speaking of make targets, I also think you should add one that runs lrsql with these given OIDC config vars.